types/block.go

```diff
@@ -290,21 +290,21 @@ func (h *Header) Hash() cmn.HexBytes {
 		return nil
 	}
 	return merkle.SimpleHashFromMap(map[string][]byte{
-		"ChainID":        cdcEncode(h.ChainID),
-		"Height":         cdcEncode(h.Height),
-		"Time":           cdcEncode(h.Time),
-		"NumTxs":         cdcEncode(h.NumTxs),
-		"TotalTxs":       cdcEncode(h.TotalTxs),
-		"LastBlockID":    cdcEncode(h.LastBlockID),
-		"LastCommit":     cdcEncode(h.LastCommitHash),
-		"Data":           cdcEncode(h.DataHash),
-		"Validators":     cdcEncode(h.ValidatorsHash),
-		"NextValidators": cdcEncode(h.NextValidatorsHash),
-		"App":            cdcEncode(h.AppHash),
-		"Consensus":      cdcEncode(h.ConsensusHash),
-		"Results":        cdcEncode(h.LastResultsHash),
-		"Evidence":       cdcEncode(h.EvidenceHash),
-		"Proposer":       cdcEncode(h.ProposerAddress),
+		"ChainID":            cdcEncode(h.ChainID),
+		"Height":             cdcEncode(h.Height),
+		"Time":               cdcEncode(h.Time),
+		"NumTxs":             cdcEncode(h.NumTxs),
+		"TotalTxs":           cdcEncode(h.TotalTxs),
+		"LastBlockID":        cdcEncode(h.LastBlockID),
+		"LastCommitHash":     cdcEncode(h.LastCommitHash),
```

Comment:
Note this actually shouldn't matter because the SimpleHashFromMap sorts the fields by key before building the merkle tree. So the only thing here that would cause a change is `Results -> LastResults`. So we need to document that as a breaking change as well.

That said, I'm not sure this key-based sorting is the best approach, because it locks us to these field names rather than their order. Thoughts?

Reply:
Copied to https://github.com/tendermint/tendermint/issues/1841#issuecomment-430001613